MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
+the interlude function and can in turn be passed to a Go function
+called by the interlude function.
+
+:code:`BufferTArray` is a list of local variables whose values
+should be saved, and restored to their original values when the
+transaction restarts. If the number of used array elements (the value
+returned by the :code:`GetUsed()` method) is zero, no local variables
+are saved and restored, and if the number of used array elements is 1,
+and that sole element references the string "*" all local variables
+are saved and restored.
+
+A case-insensitive value of "BA" or "BATCH" for :code:`transid`
+indicates to YottaDB that it need not ensure Durability for this
+transaction (it continues to ensure Atomicity, Consistency, and
+Isolation), as discussed under `ydb_tp_s()`_.
+
+A special note: as the definition and implementation of Go protect
+against dangling pointers in pure Go code, Go application code may not
+be designed and coded with the same level of defensiveness against
+dangling pointers that C applications are. In the case of
+:code:`TpS()`, owing the need to use :code:`unsafe pointer`
+parameters, please take additional care in designing and coding your
+application to ensure the validity of the pointers passed to
+:code:`TpS()`.
+
+--------------------------
+Go Simple API KeyT Methods
+--------------------------
+
+Go DataS()
+----------
+
+.. code-block:: go
+
+    DataS() int, int
+
+The :code:`DataS()` method returns the result of `ydb_data_s()`_ of
+the node referenced by the :code:`KeyT` structure, with a return code
+of :code:`C.YDB_ERR_OK` or an `error return code`_ prefixed with
+:code:`C.`. In the latter case, the return value is unspecified.
+
+Go DeleteS()
+------------
+
+.. code-block:: go
+
+    DeleteS(deltype int) int
+
+The :code:`DeleteS()` method wraps `ydb_delete_s()`_ to delete the
+local or global variable tree referenced by the :code:`KeyT`
+structure, with a value of :code:`C.YDB_DEL_NODE` for :code:`deltype`
+specifying that only the node should be deleted, leaving the tree
+untouched, and a value of :code:`C.YDB_DEL_TREE` specifying that the
+node as well as the tree are to be deleted. The return code is
+:code:`C.YDB_OK` or an `error return code`_ prefixed with :code:`C.`.
+
+Go GetValS()
+------------
+
+.. code-block:: go
+
+    GetValS(retval *BufferT) int
+
+The :code:`GetValS()` method wraps `ydb_get_s()`_ to return the value
+at the referenced global or local variable node, or intrinsic special
+variable, in the :code:`BufferT` structure referenced by
+:code:`retval`. Return codes are :code:`C.YDB_OK`,
+:code:`C.YDB_ERR_GVUNDEF`, :code:`C.YDB_ERR_INVSVN`,
+:code:`C.YDB_ERR_LVUNDEF`, or another `error return code`_ prefixed
+with :code:`C.`. See `ydb_get_s()`_ for more detailed information.
+
+Go IncrS()
+----------
+
+.. code-block:: go
+
+    IncrS(incr, retval *BufferT) int
+
+The :code:`IncrS()` method wraps `ydb_incr_s()`_ to atomically
```

Comment:
If incr is either unallocated or is nil, it takes on the default value of 1.

Reply:
Added: If :code:`incr` is unallocated or the :code:`len_used`field is zero, the default increment is 1.